### ISSUE TITLE
aws_network_acl + aws_security_group IPv6-ICMP support

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#722]
_This issue was originally opened by @bhuisgen as hashicorp/terraform#14014. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Terraform Version

0.9.3 + 0.9.4

### Affected Resource(s)

- aws_network_acl
- aws_security_group

### Terraform Configuration Files

```hcl
resource "aws_network_acl" "public" {
    ingress {
        rule_no = 2
        protocol = "icmp"
        icmp_type = -1
        icmp_code = -1
        from_port = 0
        to_port = 0
        ipv6_cidr_block = "::/0"
        action = "allow"
    }
```

```hcl
resource "aws_security_group" "machine" {
    ingress {
        protocol = "icmp"
        from_port = -1
        to_port = -1
        cidr_blocks = [
            "0.0.0.0/0"
        ]
        ipv6_cidr_blocks = [
            "::/0"
        ]
    }
```

### Debug Output

https://gist.github.com/bhuisgen/4d49a8e05a49d4e760ae3e7a9eae3ca6
### Expected Behavior

A network ACL rule with a IPv6 CIDR block and protocol set to "icmp" or 58 should create the right rule.

A security group rule with a  IPv4 + IPv6 CIDR blocks and protocol set to "icmp" or 58 should create the right rule.

### Actual Behavior

The network acl rule is created protocol ICMPv4.
The network acl rule is not created (error applying plan) if the protocol is set to 58.

The security group rule is created with ICMPv4 for the IPv6 CIDR block

### Steps to Reproduce

1. `terraform apply`

**Comment by njam** [comment]
The AWS API behaves about strange regarding this topic, and their web interface seems not 100% in line with the backend APIs.

To create a IPv6-ICMP ingress rule the *protocol* needs to be changed. The following works fine for me to allow ICMP on both IPv4 and IPv6 with terraform v0.10.8:
```hcl
  ingress {
    description = "Allow ICMP IPv4"
    from_port = -1
    to_port = -1
    protocol = "icmp"
    cidr_blocks = ["0.0.0.0/0"]
  }
  ingress {
    description = "Allow ICMP IPv6"
    from_port = -1
    to_port = -1
    protocol = "icmpv6"
    ipv6_cidr_blocks = ["::/0"]
  }
```

I'm not sure if there is a terraform bug, or just documentation should be improved?

**Comment by bflad** [comment]
Closed via #9569
